frontend/src/components/metafilter.js

```diff
@@ -0,0 +1,222 @@
+import React, { useContext, useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+
+import {
+  Badge,
+  Button,
+  Chip,
+  ChipGroup,
+  Flex,
+  FlexItem,
+} from '@patternfly/react-core';
+import {
+  Select,
+  SelectOption,
+  SelectVariant
+} from '@patternfly/react-core/deprecated';
+
+
+import { Settings } from '../settings';
+import { HttpClient } from '../services/http';
+import { toAPIFilter } from '../utilities';
+
+import { IbutsuContext } from '../services/context';
+
+// TODO Extend this to contain the filter handling functions, and better integrate filter state
+// with FilterTable. See https://github.com/ibutsu/ibutsu-server/issues/230
+const MetaFilter = (props) => {
+  const {
+    setFilter,
+    activeFilters,
+    hideFilters,
+    onRemoveFilter,
+    onApplyReport,
+    runId,
+    id
+  } = props;
+  const context = useContext(IbutsuContext);
+  const {primaryObject} = context;
+
+  const [fieldSelection, setFieldSelection] = useState([]);
+  const [isFieldOpen, setisFieldOpen] = useState(false);
+  const [isValueOpen, setIsValueOpen] = useState(false);
+
+  const [valueOptions, setValueOptions] = useState([]);
+  const [valueSelections, setValueSelections] = useState([]);
+  const [fieldOptions, setFieldOptions] = useState([]);
+
+  const onFieldSelect = (_event, selection) => {
+    // clear value state too, otherwise the old selection remains selected but is no longer visible
+    setFieldSelection(selection);
+    setisFieldOpen(false);
+    setValueSelections([]);
+    setValueOptions([]);
+    setIsValueOpen(false);
+
+  };
+
+  const onValueSelect = (_event, selection) => {
+    // update state and call setFilter
+    let updatedValues = (valueSelections.includes(selection))
+      ? valueSelections.filter(item => item !== selection)
+      : [...valueSelections, selection];
+
+    setValueSelections(updatedValues);
+    setIsValueOpen(false);
+    if (updatedValues.length > 0) {
+      setFilter(id, fieldSelection, updatedValues.join(';'));
+    }
+
+    // after filter was applied, remove the values from the fields
+    setValueSelections([]);
```

Comment:
tbh it looks a bit weird that we setValueSelection couple rows above and then clearing it. I was following behavior and I got that after you select filter and value - it's set and chip appears but select clears itself (that's fine) but we maybe don't need all of this set thing here?

Reply:
Without setting it back to an empty list the value chip stays in the dropdown (even though it's been added to the applied filters) - maybe I'm missing your suggestion for where you'd prefer to see it?

Reply:
@mshriver maybe we don't need to have `setValueSelections(updatedValues);` couple lines above

Reply:
Talked more in slack - @LightOfHeaven1994 is correct that we don't actually need to track or set the `selections` prop because on value selection the filter is applied and the select is cleared anyway.